yubihsm/objects.py

```diff
@@ -518,6 +526,12 @@ def put(
         :return: A reference to the newly created object.
         """
         if isinstance(key, rsa.RSAPrivateKeyWithSerialization):
+            public_key = key.public_key()
+            public_numbers = public_key.public_numbers()
```

Comment:
Might be a little cleaner to use rsa_numbers.public_numbers instead of via the public key.

Reply:
Agreed! I pushed a new commit that extracts the public_numbers from rsa_numbers instead.